weave/trace/client_progress_bar.py

```diff
@@ -0,0 +1,122 @@
+"""Progress bar utilities for WeaveClient.
+
+This module provides functionality for displaying progress bars when flushing
+tasks in the WeaveClient.
+"""
+
+from typing import Callable, TypedDict
+
+from rich.console import Console
+from rich.progress import (
+    BarColumn,
+    Progress,
+    SpinnerColumn,
+    TaskProgressColumn,
+    TextColumn,
+    TimeElapsedColumn,
+)
+
+
+class WeaveFlushStatus(TypedDict):
+    """Status information about the current flush operation."""
+
+    # Current job counts
+    main_jobs: int
+    fastlane_jobs: int
+    call_processor_jobs: int
+    total_jobs: int
+
+    # Tracking of completed jobs
+    completed_since_last_update: int
+    total_completed: int
+
+    # Maximum number of jobs seen during this flush operation
+    max_total_jobs: int
+
+    # Whether there are any pending jobs
+    has_pending_jobs: bool
+
+
+def create_progress_bar_callback() -> Callable[[WeaveFlushStatus], None]:
+    """Create a callback function that displays a progress bar for flush status.
+
+    Returns:
+        A callback function that can be passed to WeaveClient._flush.
+    """
+    console = Console()
```

Comment:
Did you check that it doesn't overwrite user logs?  I'm not sure why but rich sometimes paints over my terminal which can be annoying

Reply:
I think its fine? How exactly can I test that? 
<img width="1716" alt="Screenshot 2025-03-03 at 7 03 43 PM" src="https://github.com/user-attachments/assets/8a65e851-7e46-4ceb-818f-38e5bd61ac63" />

Reply:
The trace donut link gets printed while the bar is going, I think its fine.